controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -59,13 +59,9 @@ func GetClient(ctx context.Context, client resource.APIPatchingApplicator) (Gite
 	if singleInstance == nil {
 		lock.Lock()
 		defer lock.Unlock()
-		if singleInstance == nil {
```

Comment:
Oh perfect, got it, thanks!

Reply:
> Oh perfect, got it, thanks!

@kushnaidu You may want to instead update the file with comments on why the code is structured the way it is, so that others may not accidentally think it is dead code. It will also be a good first PR contribution (only if it makes sense to you) from your side.

Reply:
sure, I'll do that.

Reply:
Hey @vjayaramrh, does it look better now?